File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/null_object/solution/ternary.cpp
// ternary.cpp

class expression;
#include "ternary.h"

ternary::ternary(expression * a_operand1, expression * a_operand2, expression * a_operand3)
{
	set_operand1(a_operand1);
	set_operand2(a_operand2);
	set_operand3(a_operand3);
}

void ternary::set_operand1(expression * a_operand1)
{
	my_operand1 = a_operand1;
}

expression * ternary::get_operand1() const
{
	return my_operand1;
}

void ternary::set_operand2(expression * a_operand2)
{
	my_operand2 = a_operand2;
}

expression * ternary::get_operand2() const
{
	return my_operand2;
}

void ternary::set_operand3(expression * a_operand3)
{
	my_operand3 = a_operand3;
}

expression * ternary::get_operand3() const
{
	return my_operand3;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/null_object/solution/Ternary.java
// Ternary.java

public abstract class Ternary extends Operator
{
	public Ternary(Expression aOperand1, Expression aOperand2, Expression aOperand3)
	{
		setOperand1(aOperand1);
		setOperand2(aOperand2);
		setOperand3(aOperand3);
	}

	public abstract void accept(ExpressionVisitor aExpressionVisitor);

	private void setOperand1(Expression aOperand1)
	{
		myOperand1 = aOperand1;
	}

	public Expression getOperand1()
	{
		return myOperand1;
	}

	private void setOperand2(Expression aOperand2)
	{
		myOperand2 = aOperand2;
	}

	public Expression getOperand2()
	{
		return myOperand2;
	}

	private void setOperand3(Expression aOperand3)
	{
		myOperand3 = aOperand3;
	}

	public Expression getOperand3()
	{
		return myOperand3;
	}

	private Expression myOperand1;
	private Expression myOperand2;
	private Expression myOperand3;
}
